src/lib/containers/discussion_forum/ForumTable.tsx

```diff
@@ -205,13 +224,34 @@ export const ForumTable: React.FC<ForumTableProps> = ({
           })}
         </tbody>
       </Table>
-      {hasNextPage ? (
-        <Button variant="outline-primary" onClick={() => fetchNextPage()}>
+      {hasNextPage && (
+        <Button
+          variant="outline-primary"
+          onClick={() => {
+            fetchNextPage()
+          }}
+        >
           Show more results
         </Button>
-      ) : (
-        <></>
       )}
+      <Modal
+        size="lg"
+        show={threadModal}
+        onHide={() => setThreadModal(false)}
+        animation={false}
+      >
+        <Modal.Header>
+          <Modal.Title>New Thread</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <MarkdownEditor
+            onSave={onSave}
+            forumId={forumId}
+            onCancel={() => setThreadModal(false)}
+          />
```

Comment:
This should use a `ForumThreadEditor` component. See comments on MarkdownEditor.tsx for more details